src/common/datasource/src/file_format/csv.rs

```diff
@@ -30,14 +32,42 @@ use crate::compression::CompressionType;
 use crate::error::{self, Result};
 use crate::file_format::{self, open_with_decoder, FileFormat};
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct CsvFormat {
     pub has_header: bool,
     pub delimiter: u8,
     pub schema_infer_max_record: Option<usize>,
     pub compression_type: CompressionType,
 }
 
+impl TryFrom<&HashMap<String, String>> for CsvFormat {
+    type Error = error::Error;
+
+    fn try_from(value: &HashMap<String, String>) -> Result<Self> {
+        let mut format = CsvFormat::default();
+        if let Some(delimiter) = value.get(file_format::FORMAT_DELIMTERL) {
+            // TODO(weny): considers to support parse like "\t" (not only b'\t')
+            format.delimiter = u8::from_str(delimiter).context(error::ParseDelimiterSnafu)?;
```

Comment:
question: Should we use char for the delimiter type?

Reply:
The upstream `CsvDecoder` accepts a u8-type as a delimiter augment. However, in our case, the `sql` parser will treat input as a string(i.g., "/t", '/t'). I will double-check this later.